light/store/db/db.go

```diff
@@ -238,37 +243,30 @@ func (s *dbs) Prune(size uint16) error {
 	b := s.db.NewBatch()
 	defer b.Close()
 
-	pruned := 0
 	for itr.Valid() && numToPrune > 0 {
-		key := itr.Key()
-		height, err := s.decodeLbKey(key)
-		if err != nil {
-			return err
-		}
-		if err = b.Delete(s.lbKey(height)); err != nil {
-			return err
+		if err = b.Delete(itr.Key()); err != nil {
+			return
 		}
 		itr.Next()
 		numToPrune--
-		pruned++
 	}
 	if err = itr.Error(); err != nil {
-		return err
+		return
 	}
 
-	err = b.WriteSync()
-	if err != nil {
-		return err
-	}
-
-	// 3) Update size.
+	// 3) // update size
 	s.mtx.Lock()
-	defer s.mtx.Unlock()
+	s.size = size
+	s.mtx.Unlock()
 
-	s.size -= uint16(pruned)
+	if wErr := b.Set(s.sizeKey(), marshalSize(size)); wErr != nil {
+		err = fmt.Errorf("failed to persist size: %w", wErr)
```

Comment:
Should we not return this error? We're replacing it on the next line below.

Reply:
yup good catch